### PR TITLE
Checkbox: Added labelDisplay prop

### PR DESCRIPTION
**Pull request by ponori**
Added a `labelDisplay` prop to Checkbox along with a variant example in the docs

### CODE REVIEW

File: docs/pages/checkbox.js [hunk 1]
```diff
@@ -757,6 +757,88 @@ function CheckboxExample() {
     </Fieldset>
   );
 }
+`}
+          />
+        </MainSection.Subsection>
+
+        <MainSection.Subsection
+          title="Label visibility"
```
Comment (ayeshakmaz) [inline docs/pages/checkbox.js:765]: Small nit to add a line to the "labels" section in Accessibility about this example

File: docs/pages/checkbox.js [hunk 2]
```diff
@@ -757,6 +757,88 @@ function CheckboxExample() {
...
+          description={`In some cases, the label for a Checkbox is represented in a different way visually, as demonstrated below. In these instances, you can set labelDisplay="hidden" to ensure Checkbox is properly labeled for screen readers while using a different element to represent the label visually.`}
```
Comment (ayeshakmaz) [inline docs/pages/checkbox.js:766]: nit: wrap in back-ticks  \`labelDisplay="hidden"\`

Comment (ayeshakmaz) [reply docs/pages/checkbox.js:766]: to get the styling

**Review by ayeshakmaz — APPROVED** [review]
Approved, with two nits on the docs - thank you for adding this!

File: docs/pages/checkbox.js (hunk 2 above)
Comment (ponori) [reply docs/pages/checkbox.js:766]: Done

**Pull request merged**